TwitchLib.Client/TwitchClient.cs

```diff
@@ -1124,13 +1124,13 @@ private void JoinChannelTimeout(object sender, System.Timers.ElapsedEventArgs e)
         private Task HandleIrcMessageAsync(IrcMessage ircMessage)
         {
             var rawMessage = ircMessage.ToString();
-            if (rawMessage.StartsWith(":tmi.twitch.tv NOTICE * :Login authentication failed")
-                && OnIncorrectLogin != null)
-            {
-                return OnIncorrectLogin.Invoke(this, new()
+            if (rawMessage.StartsWith(":tmi.twitch.tv NOTICE * :Login authentication failed"))
+            { 
+                // Does this need to fallback to UnaccountedFor?
+                return OnIncorrectLogin?.Invoke(this, new()
```

Comment:
Can you null check a Task like this? Iirc you cant.
This is why I made an extension method for the EventSub packages in their shared package that does the check otherwise to prevent that.
CC: https://github.com/TwitchLib/TwitchLib.EventSub.Core/blob/main/TwitchLib.EventSub.Core/Extensions/AsyncEventHandlerExtensions.cs

Reply:
The code in your helper is functionally equivalent to this one, since I'm not awaiting a task (which would throw) but returning it instead (we do not need to force await it here, since the method only does pre-processing).

Reply:
yeah my bad I didnt think about it that way. the think I tried to fix was basically awaiting null but that isnt the case here!